pkg/oam/util/helper.go

```diff
@@ -367,32 +367,31 @@ func GetCapabilityDefinition(ctx context.Context, cli client.Reader, definition
 }
 
 func fetchDefinitionRev(ctx context.Context, cli client.Reader, definitionName string) (bool, *v1beta1.DefinitionRevision, error) {
-	defRevName, err := ConvertDefinitionRevName(definitionName)
-	if err != nil {
-		if err.Error() == ErrBadRevision {
-			return true, nil, nil
-		}
-		return false, nil, err
+	// if the component's type doesn't contain '@' means user want to use the latest Definition.
+	if !strings.Contains(definitionName, "@") {
+		return true, nil, nil
 	}
+
+	defRevName := ConvertDefinitionRevName(definitionName)
 	defRev := new(v1beta1.DefinitionRevision)
-	if err = GetDefinition(ctx, cli, defRev, defRevName); err != nil {
+	if err := GetDefinition(ctx, cli, defRev, defRevName); err != nil {
 		return false, nil, err
 	}
-	return false, defRev, err
+	return false, defRev, nil
 }
 
 // ConvertDefinitionRevName can help convert definition type defined in Application to DefinitionRevision Name
-// e.g., worker@v2 will be convert to worker-v2
-func ConvertDefinitionRevName(definitionName string) (string, error) {
-	revNum, err := ExtractRevisionNum(definitionName, "@")
-	if err != nil {
-		return "", err
-	}
-	defName := strings.TrimSuffix(definitionName, fmt.Sprintf("@v%d", revNum))
-	if defName == "" {
-		return "", fmt.Errorf("invalid definition defName %s", definitionName)
+// e.g., worker@v1.3.1 will be convert to worker-v1.3.1
+func ConvertDefinitionRevName(definitionName string) string {
+	splits := strings.Split(definitionName, "@v")
+	if len(splits) == 1 || len(splits[0]) == 0 {
+		return definitionName
 	}
-	return fmt.Sprintf("%s-v%d", defName, revNum), nil
+
+	defName := splits[0]
+	revisionName := strings.TrimPrefix(definitionName, fmt.Sprintf("%s@v", defName))
+
+	return fmt.Sprintf("%s-v%s", defName, revisionName)
```

Comment:
you should validate the final name

Reply:
or add validation in webhook, then there's no need

Reply:
sure, we must check the final name of definitionRevision, i will validate the name in webhook